### PR TITLE
Color Picker Plugin

### LINKED ISSUE #410: Color picker plugin
https://discordjs.guide/popular-topics/canvas.html#getting-started
Use Canvas to send an image of just the target color

### PR DESCRIPTION
**Pull request by tanndlin**
fixes #410

### CODE REVIEW

File: src/app/plugins/colorpicker.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  public description: string = 'Sends an image of the desired color';
+  public usage: string = 'colorpicker #<hex color>';
+  public pluginAlias = ['cp'];
+  public permission: ChannelType = ChannelType.Public;
+  public commandPattern: RegExp = /#[0-9a-f]{6}/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
```
Comment (joey-colon) [inline src/app/plugins/colorpicker.plugin.ts:17]: why no validate method?

Comment (tanndlin) [reply src/app/plugins/colorpicker.plugin.ts:17]: I refactored it a little bit sometime ago
 The default [validate](https://github.com/cs-discord-at-ucf/lion/blob/6fff4915afaa7cbf9739b0688ab4ba3ef988ae95/src/common/plugin.ts#L29-L32) takes in a regex uses to test the args

Comment (tanndlin) [reply src/app/plugins/colorpicker.plugin.ts:17]: Regex is empty by default, so it returns true unless you give a public `commandPattern` or you can override it

Comment (joey-colon) [reply src/app/plugins/colorpicker.plugin.ts:17]: ahh was not aware. nice

**Review by joey-colon — APPROVED** [review]

**Pull request merged**